GithubRepo/Scenes/Search/View/SearchViewController.swift

```diff
@@ -0,0 +1,177 @@
+        
+        handleSearchInitialState()
+        
+        view.addSubview(spinner)
+        spinner.translatesAutoresizingMaskIntoConstraints = false
+        spinner.centerXAnchor.constraint(equalTo: view.centerXAnchor).isActive = true
+        spinner.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor, constant: 16).isActive = true
+    }
+    
+    
+    private func registerTableViewCell() {
+        tableView?.delegate = self
+        tableView?.dataSource = self
+        
+        tableView?.register(UINib(nibName: kRepositoryTableViewCellIdentifier, bundle: nil), forCellReuseIdentifier: kRepositoryTableViewCellIdentifier)
+    }
+    
+    // MARK: - Helper Methods
+    
+    private func handleSearchInitialState() {
+        self.stateView?.isHidden = false
+        self.stateTextView?.text = kInitialSearchStateText
+        self.stateImageView?.image = #imageLiteral(resourceName: "Bookmark")
+    }
+    
+    private func handleSearchEmptyState() {
+        self.stateView?.isHidden = false
+        self.stateTextView?.text = kEmptySearchStateText
+        self.stateImageView?.image = #imageLiteral(resourceName: "BookmarkMad")
+    }
+    
+    private func handleSearchErrorState() {
+        self.stateView?.isHidden = false
+        self.stateTextView?.text = kErrorSearchStateText
+        self.stateImageView?.image = #imageLiteral(resourceName: "BookmarkError")
+    }
+}
+
+// MARK: - Notifications from View Model
+
+extension SearchViewController: SearchViewModelDelegate {
+    func onChangeSearchError(error: Bool) {
+        DispatchQueue.main.async {
+            if error {
+                self.handleSearchErrorState()
+            }
+        }
+    }
+    
+    func onChangeSearchLoadingState(isLoading: Bool) {
+        DispatchQueue.main.async {
+            if isLoading {
+                self.spinner.startAnimating()
+            } else {
+                self.spinner.stopAnimating()
+            }
+        }
+    }
+    
+    func onChangeSearchRepository(repoCellViewModels: [RepositoryCellViewModel]) {
+        DispatchQueue.main.async {
+            if repoCellViewModels.isEmpty {
+                self.handleSearchEmptyState()
+            } else {
+                self.stateView?.isHidden = true
+                self.tableView?.reloadData()
+            }
+        }
+    }
+}
+
+// MARK: - Table View Extension
+
+extension SearchViewController: UITableViewDataSource, UITableViewDelegate {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return searchViewModel.repositoryCellViewModels.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: kRepositoryTableViewCellIdentifier) as? RepositoryTableViewCell, indexPath.row < self.searchViewModel.repositoryCellViewModels.count else {
```

Comment:
check line length. We usually like to use `120` as the limit. you can set a vertical guide line in your xcode